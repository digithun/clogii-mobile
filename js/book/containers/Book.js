import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import _ from 'lodash';

import Book from '../components/Book';

export const query = gql`
  query BookQuery($id: MongoID!){
    clog(filter: { _id: $id }) {
      ...Book
    }
  }
  ${Book.fragments.clog}
`;

export const mapQueryToProps = ({ data }) => {
  const { clog, error, loading } = data;
  if (error) {
    console.error('graphql error: ', error);
  }
  return {
    clog: {
      ...(clog || {}),
      episodes: loading || error ? [] : _.sortBy(clog.episodes, ep => -ep.no),
    },
    loading,
  };
};

export const mapPropsToOptions = ({ id }) => ({
  variables: {
    id,
  },
});

export default graphql(query, {
  props: mapQueryToProps,
  options: mapPropsToOptions,
})(Book);

// export default require('View')
