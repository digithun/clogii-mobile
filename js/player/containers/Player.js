import Player from '../components/Player';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';

export const query = gql`
  query PlayerQuery($id: ID!){
    episode(id: $id) {
      ...Player
    }
  }
  ${Player.fragments.episode}
`;

export const mapQueryToProps = ({ data }) => {
  const {episode, error, loading} = data;
  if (error) {
    console.error('graphql error: ', error);
  }
  return {
    loading: loading,
    episode: episode || {}
  };
}
export const mapPropsToOptions = ({id}) => ({
  variables: {
    id
  }
});

export default graphql(query, {
  props: mapQueryToProps,
  options: mapPropsToOptions
})(Player);
