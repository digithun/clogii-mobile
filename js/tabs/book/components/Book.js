import React from 'react';
import {
  View,
  Image,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity
} from 'react-native';
import moment from 'moment';

import NavBar from '../../../common/NavBar';
import BorderButton, {styles as borderButtonStyles} from '../../../common/BorderButton';
import PureListView from '../../../common/PureListView';
import CircleImage from '../../../common/CircleImage';
import {toHumanNumber, mapSource} from '../../../common/utils';
import {colors, styles as commonStyles} from '../../../common/styles';

const MetaEpisode = (props) => (
  <View style={styles.metaEpisodeContainer}>
    <View style={{width: 60}}>
      <CircleImage
        source={mapSource(props.preview)}
        size={50}
        />
    </View>
    <View style={{width: 200}}>
      <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>
        <View style={{flexDirection: 'row', alignItems: 'center'}}>
          <Text style={styles.textEpisodeNo}>ตอนที่ </Text>
          <Text style={styles.textEpisodeNo}>{props.no}</Text>
        </View>
        <View style={{}}>
          <Text style={styles.textEpisodeUpdateAt}>{moment(props.createdAt).locale('en').format('MMMM D')}</Text>
        </View>
      </View>
      <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}>
        <View style={{flex: 3, flexDirection: 'row', alignItems: 'center'}}>
          <Image source={require('../../../common/img/icon/read.png')} style={{width: 20, height: 20, resizeMode: 'contain', borderRadius: 4}}/>
          <Text style={{paddingLeft: 5, fontSize: 12, color: colors.textFadedGrey}}>ดู {toHumanNumber(props.viewCount || 0)} ครั้ง</Text>
        </View>
        <View style={{flex: 4, flexDirection: 'row', alignItems: 'center'}}>
          <Image source={require('../../../common/img/icon/heart.png')} style={{width: 15, height: 15, resizeMode: 'contain', borderRadius: 4, marginLeft: 20}}/>
          <Text style={{paddingLeft: 5, fontSize: 12, color: colors.textFadedGrey}}>{toHumanNumber(props.likeCount || 0)} Like</Text>
        </View>
      </View>
    </View>
    <View style={{flex: 1, alignItems: 'flex-end', paddingLeft: 10}}>
      {
        !props.lock ?
        <BorderButton containerStyle={styles.metaEpisodeButton} type="borderFadedBlack" caption="อ่าน"/>
        :
        <BorderButton containerStyle={styles.metaEpisodeButton} textStyle={styles.unlockEpisodeText} type="lightGreen" caption="jelly"
          renderBeforeText={() => <LockImg/>}/>
      }
    </View>
  </View>
);

const LockImg = () => (
  <View style={{flexDirection: 'row', marginRight: 2, alignItems: 'center', justifyContent: 'center'}}>
    <Image style={{height: 10, width: 8, resizeMode: 'stretch', marginRight: 2}} source={require('../../img/icons/lock.png')}/>
    <Text style={[{fontSize: 12, fontWeight: 'bold', lineHeight: 12}, borderButtonStyles.lightGreenText]}>20</Text>
  </View>
);

const SubDetail = ({title, author, review }) => (
  <View style={styles.subDetailContainer}>
    <View style={{flexDirection: 'row', paddingVertical: 10}}>
      <View style={{flex: 3}}>
        <View>
          <Text style={styles.titleText}>{title}</Text>
          <Text style={styles.authorText}>{author.name}</Text>
        </View>
      </View>
      <View style={{paddingHorizontal: 5}}>
        <BorderButton type="lightGreen" caption="ติดตาม"/>
      </View>
    </View>
    <View>
      <View style={styles.reviewContainer}>
        <Text style={styles.reviewText}>{review}</Text>
      </View>
    </View>
    <View>
      <View style={styles.startReadButtonContainer}>
        <BorderButton
          type="borderFadedBlack"
          caption="เริ่มอ่านตอนที่ 1"
          textStyle={{
            fontSize: 25
          }}
          containerStyle={{
            paddingVertical: 5,
            justifyContent: 'center',
            alignItems: 'center'
          }}
          />
      </View>
    </View>
  </View>
);

class Book extends React.Component {
  render() {
    if (this.props.loading) {
      return null;
    }
    const clog = this.props.clog;
    return (
      <View style={{flex: 1}}>
        <Image source={require('../img/mockCover.png')} style={styles.cover}>
          <TouchableOpacity style={styles.backButtonContainer} onPress={this.onBackPress.bind(this)}>
            <Image source={require('../../../common/img/icon/backButton.png')} style={commonStyles.navBarIcon}/>
          </TouchableOpacity>
        </Image>
        <ScrollView style={styles.detailContainer}>
          <SubDetail {...clog}/>
          <View style={styles.episodeContainer}>
            <View style={{height: 1, backgroundColor: colors.greyBorder}}/>
            {
              clog.episodes.map(this.renderEpisode.bind(this))
            }
          </View>
        </ScrollView>
      </View>
    );
  }

  renderEpisode(data) {
    return (
      <View>
        <MetaEpisode {...data}/>
        <View style={{height: 1, backgroundColor: colors.greyBorder}}/>
      </View>
    );
  }

  onBackPress() {
    this.props.onBackPress && this.props.onBackPress();
  }
}

const styles = StyleSheet.create({
  cover: {
    height: 200,
    width: undefined,
    resizeMode: 'cover'
  },
  backButtonContainer: {
    top: 20,
    left: 10
  },
  detailContainer: {
    flex: 1,
    padding: 15
  },
  subDetailContainer: {
    width: undefined
  },
  titleText: {
    fontSize: 18
  },
  authorText: {
    fontSize: 12,
    color: colors.textFadedGrey
  },
  reviewContainer: {},
  reviewText: {
    fontSize: 14,
    color: colors.textGrey
  },
  startReadButtonContainer: {
    paddingVertical: 15,
    paddingHorizontal: 40
  },
  episodeContainer: {
    paddingVertical: 10
  },
  metaEpisodeContainer: {
    paddingVertical: 8,
    paddingHorizontal: 5,
    flexDirection: 'row',
    alignItems: 'center'
  },
  metaEpisodeButton: {
    paddingVertical: 2,
    width: 60,
    alignItems: 'center'
  },
  textEpisodeNo: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  textEpisodeUpdateAt: {
    fontSize: 10,
    paddingLeft: 5,
    color: colors.textFadedGrey
  },
  unlockEpisodeText: {
    fontSize: 10
  }
});

export default Book;
