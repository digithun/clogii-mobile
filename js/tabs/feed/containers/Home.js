import React from 'react';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';

import FeedHome from '../components/FeedHome';
// import {CLOG_PREVIEW_LIMIT} from '../constants';

import { fragments } from '../../../models/clog';

export const query = gql`
    query {
        trendingClogs: clogs(filter: {limit: 20}, orderBy: TRENDING) {
            ...clogMetaData
        }
        recommendedClog {
            ...clogMetaData
            review
        }
        favoriteTags {
            name
            trendingClogs {
                ...clogMetaData
            }
        }
        heroBanners {
            ...clogMetaData
            review
        }
    }
    ${fragments.clogMetaData}
`;

export const mapQueryToProps = ({ ownProps, data }) => {
  const { loading, trendingClogs, recommendedClog, favoriteTags, heroBanners } = data;
  return ({
    trendingClogs: loading ? [] : trendingClogs,
    recommendedClog,
    heroBanners: loading ? [] : heroBanners,
    favoriteTags: loading ? [] : favoriteTags,
    loading,
  });
};

export default graphql(query, {
  props: mapQueryToProps,
})(FeedHome);
