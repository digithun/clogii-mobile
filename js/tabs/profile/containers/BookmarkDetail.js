import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import BookmarkDetail from '../components/BookmarkDetail';
import * as mockData from '../mockData';

const query = gql`
  query BookmarkDetail($id: MongoID!){
    clog(filter: { _id: $id }) {
      ...BookmarkDetail
    }
  }
  ${BookmarkDetail.fragments.clog}
`;

const mapQueryToProps = ({ data }) => {
  return {
    clog: data.clog,
  };
};

export const mapPropsToOptions = ({ id }) => ({
  variables: {
    id,
  },
});

export default graphql(
  query,
  {
    props: mapQueryToProps,
    options: mapPropsToOptions,
  },
)(BookmarkDetail);
