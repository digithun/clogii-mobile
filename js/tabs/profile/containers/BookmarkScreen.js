import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import BookmarkScreen from '../components/BookmarkScreen';
import * as mockData from '../mockData';

export const query = gql`
  query {
    me {
      id
      summaryBookmarks {
        ...BookmarkScreen
      }
    }
  }
  ${BookmarkScreen.fragments.summaryBookmark}
`;

const mapQueryToProps = ({ data }) => {
  return {
    summaryBookmarks: !data.me || !data.me.summaryBookmarks ? [] : data.me.summaryBookmarks,
  };
};

export default graphql(
  query,
  {
    props: mapQueryToProps,
  }
)(BookmarkScreen);
