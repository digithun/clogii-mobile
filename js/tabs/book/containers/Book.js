import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import _ from 'lodash';

import Book from '../components/Book';

export const query = gql`
  query Clog($id: ID!){
    clog(id: $id) {
      title
      cover
      review
      author {
        name
      }
      episodes {
        no
        preview
        viewCount
        likeCount
        createdAt
      }
    }
  }
`;

export const mapQueryToProps = ({ ownProps, data }) => {
  return {
    clog: {
      ...data.clog,
      episodes: data.loading ? [] : _.sortBy(data.clog.episodes, 'no')
    },
    loading: data.loading
  };
};

export const mapPropsToOptions = ({id}) => ({
  variables: {
    id
  }
});

export default graphql(query, {
  props: mapQueryToProps,
  options: mapPropsToOptions
})(Book);

// export default require('View')
