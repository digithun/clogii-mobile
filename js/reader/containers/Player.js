import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import _ from 'lodash';

import { filterClogId, filterEpisodeId, withAddEpisodeBookmark, withRemoveBookmarks, updateMeBookmarksReduer } from '../../models/bookmark';
import Player from '../components/Player';

export const query = gql`
  query PlayerQuery($id: MongoID!){
    episode(filter: { _id: $id }) {
      ...PlayerEpisode
    }
    me {
      id
      bookmarks {
        ...PlayerBookmark
      }
    }
  }
  ${Player.fragments.episode}
  ${Player.fragments.bookmark}
`;

export const mapQueryToProps = ({ data }) => {
  const { episode, error, loading, me } = data;
  if (error) {
    console.error('graphql error: ', error);
  }
  let episodeBookmarks;
  if (episode && me) {
    episodeBookmarks = filterEpisodeId(filterClogId(me.bookmarks, episode.clogId), episode.id);
  }

  return {
    loading,
    episode: episode || {},
    episodeBookmark: episodeBookmarks ? episodeBookmarks[0] : null,
  };
};
export const mapPropsToOptions = ({ id }) => ({
  variables: {
    id,
  },
  reducer: updateMeBookmarksReduer,
});

const withData = graphql(query, {
  props: mapQueryToProps,
  options: mapPropsToOptions,
});

export default withData(withRemoveBookmarks(withAddEpisodeBookmark(Player)));