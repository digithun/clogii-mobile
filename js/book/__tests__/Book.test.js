import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import graphql from '../../libs/mockGraphQL';

import Book from '../components/Book';
import { query, mapQueryToProps, mapPropsToOptions } from '../containers/Book';

function getBookProps(props) {
  return graphql(query, mapPropsToOptions(props)).then(result => mapQueryToProps({ data: result.data }));
}

describe('Book', () => {
  it('render', async () => {
    const props = { id: 1 };
    const dataProps = await getBookProps(props);
    const tree = renderer.create(<Book {...props} {...dataProps} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
