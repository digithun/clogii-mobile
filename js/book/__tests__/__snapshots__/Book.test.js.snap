exports[`Book render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <Image
    source={
      Object {
        "uri": "http://localhost:8080/static/cover/mockCover.png",
      }
    }
    style={
      Object {
        "height": 200,
        "resizeMode": "cover",
        "width": undefined,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 10,
          "opacity": 1,
          "top": 20,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
          }
        } />
    </View>
  </Image>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <ScrollView
      contentInset={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      data={
        Array [
          Object {
            "__typename": "Episode",
            "createdAt": "2005-10-26T06:56:32.695Z",
            "id": "185866782554276124377636",
            "likeCount": 100,
            "no": 11,
            "thumbnailImage": "http://localhost:8080/static/preview/g4722.png",
            "viewCount": 1258,
            Symbol(id): "Episode185866782554276124377636",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "1979-12-15T07:25:31.435Z",
            "id": "739617331384591268883859",
            "likeCount": 100,
            "no": 10,
            "thumbnailImage": "http://localhost:8080/static/preview/g3618.png",
            "viewCount": 546,
            Symbol(id): "Episode739617331384591268883859",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "1981-10-17T06:32:19.457Z",
            "id": "821682781513984683828379",
            "likeCount": 100,
            "no": 9,
            "thumbnailImage": "http://localhost:8080/static/preview/g3578.png",
            "viewCount": 825,
            Symbol(id): "Episode821682781513984683828379",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "2016-10-14T19:27:19.075Z",
            "id": "588349258616913463578285",
            "likeCount": 100,
            "no": 8,
            "thumbnailImage": "http://localhost:8080/static/preview/g4762.png",
            "viewCount": 1153,
            Symbol(id): "Episode588349258616913463578285",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "2013-02-10T20:01:37.113Z",
            "id": "371958189726321388659115",
            "likeCount": 100,
            "no": 7,
            "thumbnailImage": "http://localhost:8080/static/preview/g3618.png",
            "viewCount": 2994,
            Symbol(id): "Episode371958189726321388659115",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "2012-05-07T20:43:27.951Z",
            "id": "583747425875566218122443",
            "likeCount": 100,
            "no": 6,
            "thumbnailImage": "http://localhost:8080/static/preview/g3618.png",
            "viewCount": 319,
            Symbol(id): "Episode583747425875566218122443",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "1971-05-29T06:10:12.845Z",
            "id": "113328316263021647243388",
            "likeCount": 100,
            "no": 5,
            "thumbnailImage": "http://localhost:8080/static/preview/g3618.png",
            "viewCount": 2785,
            Symbol(id): "Episode113328316263021647243388",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "1977-02-19T10:44:23.932Z",
            "id": "035528932341474245766169",
            "likeCount": 100,
            "no": 4,
            "thumbnailImage": "http://localhost:8080/static/preview/g4602.png",
            "viewCount": 2458,
            Symbol(id): "Episode035528932341474245766169",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "1994-05-02T15:21:13.246Z",
            "id": "818143150096814770571155",
            "likeCount": 100,
            "no": 3,
            "thumbnailImage": "http://localhost:8080/static/preview/g4642.png",
            "viewCount": 2340,
            Symbol(id): "Episode818143150096814770571155",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "1998-08-10T18:37:12.677Z",
            "id": "052548234969651762031017",
            "likeCount": 100,
            "no": 2,
            "thumbnailImage": "http://localhost:8080/static/preview/g3578.png",
            "viewCount": 2801,
            Symbol(id): "Episode052548234969651762031017",
          },
          Object {
            "__typename": "Episode",
            "createdAt": "2007-08-24T23:54:36.100Z",
            "id": "262969383465034245756718",
            "likeCount": 100,
            "no": 1,
            "thumbnailImage": "http://localhost:8080/static/preview/g3336.png",
            "viewCount": 1431,
            Symbol(id): "Episode262969383465034245756718",
          },
        ]
      }
      dataSource={
        ListViewDataSource {
          "items": 11,
        }
      }
      initialListSize={1}
      minContentHeight={0}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={1000}
      onKeyboardDidHide={undefined}
      onKeyboardDidShow={undefined}
      onKeyboardWillHide={undefined}
      onKeyboardWillShow={undefined}
      onLayout={[Function]}
      onScroll={[Function]}
      pageSize={1}
      removeClippedSubviews={true}
      renderEmptyList={null}
      renderFooter={[Function]}
      renderHeader={[Function]}
      renderRow={[Function]}
      renderSeparator={[Function]}
      scrollEventThrottle={50}
      scrollRenderAheadDistance={1000}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 15,
        }
      }>
      <View>
        <View
          style={
            Object {
              "width": undefined,
            }
          }>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingVertical": 10,
              }
            }>
            <View
              style={
                Object {
                  "flex": 3,
                }
              }>
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 18,
                    }
                  }>
                  Non itaque quia
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.3)",
                      "fontSize": 12,
                    }
                  }>
                  Ms. Shirley Mayer
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingHorizontal": 5,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "rgb(141, 227, 188)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingHorizontal": 15,
                  }
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      Object {
                        "color": "rgb(141, 227, 188)",
                      },
                      undefined,
                    ]
                  }>
                  ติดตาม
                </Text>
              </View>
            </View>
          </View>
          <View>
            <View
              style={Object {}}>
              <View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={0}>
                  Aperiam quia mollitia hic. Quia illum ullam in tempora soluta reiciendis. Hic sed molestiae sed dicta reiciendis voluptatem. Amet distinctio quo corporis. Assumenda omnis accusantium possimus. Voluptatum est ratione fugiat nam. Quidem assumenda quis repudiandae est eaque consequatur. Iure molestiae aut dolores laudantium occaecati aliquam voluptatem veniam quia. Illo ipsam qui enim autem sunt aperiam. Occaecati et itaque voluptatibus voluptas voluptatum doloremque sed sequi vitae. Doloribus recusandae odit iusto quisquam laboriosam reprehenderit. Ullam nulla corrupti quo itaque voluptatem aut. Eius qui nesciunt doloribus ipsam ut dolor facilis et. Adipisci eveniet similique explicabo et minus eius ut enim optio. Qui ipsa magni est aut pariatur laudantium velit inventore quibusdam. Consequatur doloribus harum et. Delectus quia debitis quod iste omnis dolores. Id debitis sed sit adipisci nihil est dolorem. Inventore rem pariatur est eum. Dolores voluptatem omnis recusandae quo voluptatum laudantium sed qui.
                </Text>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "paddingHorizontal": 40,
                  "paddingVertical": 15,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(0, 0, 0, 0.8)",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingHorizontal": 15,
                    "paddingVertical": 5,
                  }
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      Object {
                        "color": "white",
                      },
                      Object {
                        "fontSize": 25,
                      },
                    ]
                  }>
                  เริ่มอ่านตอนที่ 1
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "height": 1,
            }
          } />
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingLeft": 5,
              "paddingVertical": 8,
            }
          }>
          <View
            style={
              Object {
                "width": 60,
              }
            }>
            <View
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }>
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0)",
                    "borderRadius": 25,
                    "height": 50,
                    "left": -0,
                    "position": "absolute",
                    "top": -0,
                    "width": 50,
                  }
                } />
              <Image
                size={50}
                source={
                  Object {
                    "uri": "http://localhost:8080/static/preview/g4722.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderRadius": 25,
                      "height": 50,
                      "width": 50,
                    },
                    undefined,
                  ]
                } />
              <View
                style={
                  Object {
                    "height": 50,
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": 50,
                  }
                } />
            </View>
          </View>
          <View
            style={
              Object {
                "width": 180,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }>
                  ตอนที่ 
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }>
                  11
                </Text>
              </View>
              <View
                style={Object {}}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.3)",
                      "fontSize": 10,
                      "paddingLeft": 5,
                    }
                  }>
                  October 26
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 3,
                    "flexDirection": "row",
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  } />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.3)",
                      "fontSize": 12,
                      "paddingLeft": 5,
                    }
                  }>
                  ดู 
                  1,258
                   ครั้ง
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 4,
                    "flexDirection": "row",
                  }
                }>
                <Image
                  source={1}
                  style={
                    Object {
                      "borderRadius": 4,
                      "height": 15,
                      "marginLeft": 20,
                      "resizeMode": "contain",
                      "width": 15,
                    }
                  } />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.3)",
                      "fontSize": 12,
                      "paddingLeft": 5,
                    }
                  }>
                  100
                   Like
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "paddingRight": 45.5,
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0, 0, 0, 0.8)",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingHorizontal": 15,
                  "paddingVertical": 1,
                  "width": 55,
                }
              }
              testID={undefined}>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 12,
                    },
                    Object {
                      "color": "white",
                    },
                    undefined,
                  ]
                }>
                อ่าน
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#eeeeee",
            "height": 1,
          }
        } />
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 45,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1.6,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
          }
        }>
        100
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1.6,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
          }
        }>
        0
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "resizeMode": "contain",
          }
        } />
    </View>
  </View>
</View>
`;
